aten/src/ATen/core/DistributionsHelper.h

```diff
@@ -140,19 +229,22 @@ struct normal_distribution {
         return ret;
       }
     }
+#endif
     // otherwise generate new normal values
     uniform_real_distribution<T> uniform(0.0, 1.0);
     const dist_acctype<T> u1 = uniform(generator);
     const dist_acctype<T> u2 = uniform(generator);
     const dist_acctype<T> r = ::sqrt(static_cast<T>(-2.0) * ::log(static_cast<T>(1.0)-u2));
     const dist_acctype<T> theta = static_cast<T>(2.0) * static_cast<T>(M_PI) * u1;
+#if !defined(__CUDACC__) && !defined(__HIPCC__)
     if (std::is_same<T, double>::value) {
       dist_acctype<double> cache = r * ::sin(theta);
       generator->set_next_double_normal_sample(c10::optional<double>(cache));
     } else {
       dist_acctype<float> cache = r * ::sin(theta);
       generator->set_next_float_normal_sample(c10::optional<float>(cache));
     }
+#endif
```

Comment:
Seeing that this code was generalized to work on CUDA, I was expecting to see some call sites to cosine in cuda files get removed. But I didn't see any while grepping. Is this actually being exercised from CUDA?

Reply:
This is used in [the next PR](https://github.com/pytorch/pytorch/pull/36633), that's why this PR is called "Prerequisites" :)